src/adaptors/mod.rs

```diff
@@ -113,21 +113,22 @@ impl<I, J> Iterator for InterleaveShortest<I, J>
 
     #[inline]
     fn next(&mut self) -> Option<Self::Item> {
-        match self.phase {
-            false => match self.it0.next() {
+        if self.phase {
+            match self.it1.next() {
                 None => None,
                 e => {
-                    self.phase = true;
+                    self.phase = false;
                     e
                 }
-            },
-            true => match self.it1.next() {
+            }
+        } else {
+            match self.it0.next() {
                 None => None,
                 e => {
-                    self.phase = false;
+                    self.phase = true;
                     e
                 }
-            },
+            }
         }
     }
```

Comment:
I think this method could be written much more succinctly as:
```rust
    #[inline]
    fn next(&mut self) -> Option<Self::Item> {
        let it = if self.phase { self.it1 } else { self.it0 };
        let e = it.next();

        if e.is_some() {
            self.phase = !self.phase;
        }

        e
    }
```

Reply:
I thought the same, but `it1` and `it0` have different types.

Reply:
D'oh. How about this, then:

```
    #[inline]
    fn next(&mut self) -> Option<Self::Item> {
        let e = if self.phase { self.it1.next() } else { self.it0.next() };

        if e.is_some() {
            self.phase = !self.phase;
        }

        e
    }
```

Reply:
Nice. I took this.